File: StructuralPattern/src/org/uncle/lee/facade/Facade.java
package org.uncle.lee.facade;

public class Facade {

	private SubSystemA systemA;
	private SubSystemB systemB;
	private SubSystemC systemC;

	public Facade(){
		systemA = new SubSystemA();
		systemB = new SubSystemB();
		systemC = new SubSystemC();
	}
	
	public void doAB(){
		systemA.operate();
		systemB.operate();
		printBlankLine();
	}
	
	public void doBC(){
		systemB.operate();
		systemC.operate();
		printBlankLine();
	}
	
	public void doCA(){
		systemC.operate();
		systemA.operate();
		printBlankLine();
	}
	
	private void printBlankLine(){
		System.out.println("\n");
	}
}

File: Kotlin/src/structural/facade/Facade.kt
package structural.facade

class Facade {
    var subSystemA : SubSystemA? = null
    var subSystemB : SubSystemB? = null
    var subSystemC : SubSystemC? = null

    fun doAB() {
        subSystemA!!.operatorA()
        subSystemB!!.operatorB()
    }

    fun doBC() {
        subSystemB!!.operatorB()
        subSystemC!!.operatorC()
    }
}